clang/test/CXX/basic/basic.lookup/basic.lookup.qual/class.qual/p2.cpp

```diff
@@ -137,17 +137,16 @@ namespace InhCtor {
   int n = b.T(); // expected-error {{'T' is a protected member of 'InhCtor::A'}}
                  // expected-note@-15 {{declared protected here}}
 
-  // FIXME: EDG and GCC reject this too, but it's not clear why it would be
-  // ill-formed.
   template<typename T>
   struct S : T {
-    struct U : S { // expected-note 6{{candidate}}
+    // FIXME: S is incomplete here and we should diagnose this!
+    struct U : S {
```

Comment:
Did the FIXME ever get a core issue?  Perhaps we need to run that down a bit?  @Endilll or @shafik , any chance the 'FIXME" that we just 'fixed' looks familiar?

Reply:
This rings a bell. I've put something similar in our DR tests, probably in a PR about complete-class context. I'm running out of time today, but I'll look into this tomorrow if Shafik doesn't beat me to it.

Reply:
The FIXME wasn't fixed, it's just broken in a different way that's consistent with other "current instantiation broken-ness" related things. I'm going to submit a patch (probably this week) which diagnoses cases where the current instantiation is used in a context which requires a complete type.

Reply:
There are somewhat related CWG issues [45](https://cplusplus.github.io/CWG/issues/45.html), [374](https://cplusplus.github.io/CWG/issues/347.html), and [500](https://cplusplus.github.io/CWG/issues/500.html), but I can't find one arguing that base specifiers should become a complete-class context. http://eel.is/c++draft/class#mem.general-8 seems quite clear on this.